dll/ntdll/rtl/libsupp.c

```diff
@@ -1221,27 +1221,19 @@ ULONG
 NTAPI
 RtlGetTickCount(VOID)
 {
-    ULARGE_INTEGER TickCount;
+    LARGE_INTEGER TickCount;
+
+    TickCount = KiReadSystemTime(&SharedUserData->TickCount);
 
 #ifdef _WIN64
-    TickCount.QuadPart = *((volatile ULONG64*)&SharedUserData->TickCount);
+    return (TickCount.QuadPart * SharedUserData->TickCountMultiplier) >> 24;
 #else
-    while (TRUE)
-    {
-        TickCount.HighPart = (ULONG)SharedUserData->TickCount.High1Time;
-        TickCount.LowPart = SharedUserData->TickCount.LowPart;
-
-        if (TickCount.HighPart == (ULONG)SharedUserData->TickCount.High2Time)
-            break;
-
-        YieldProcessor();
-    }
-#endif
-
+    ULONG TickCountMultiplier = SharedUserData->TickCountMultiplier;
```

Comment:
I wonder if it might be possible to use `Int64ShrlMod32` intrinsic like below
https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/gdi/eng/engmisc.c;hb=2e85425c5330661dcc4fef68904e6675cafc0a6a#l298

Reply:
Or maybe we should just use native math on x86 and let the compiler do whatever it needs to do. I might do some tests. But I will not address that in this PR.